src/protocol/modulus_conversion/convert_shares.rs

```diff
@@ -108,6 +108,36 @@ impl ConvertShares {
     }
 }
 
+pub async fn convert_all_shares<F: Field>(
+    ctx: &ProtocolContext<'_, F>,
+    input: &[u64],
+) -> Result<Vec<Vec<Replicated<F>>>, BoxError> {
+    let converted_shares = try_join_all(zip(repeat(ctx), input).enumerate().map(
+        |(record_id, (ctx, row))| async move {
+            ConvertShares::new(XorShares {
+                num_bits: 64,
+                packed_bits: *row,
+            })
+            .execute(
+                ctx.narrow(&ModulusConversion(record_id.try_into().unwrap())),
+                RecordId::from(record_id),
+            )
+            .await
+        },
+    ))
+    .await?;
+
+    //Transpose the output to return bit-wise results
+    let cols = vec![Replicated::new(F::ZERO, F::ZERO); converted_shares.len()];
+    let mut output = vec![cols; converted_shares[0].len()];
+    for (row_idx, row) in converted_shares.into_iter().enumerate() {
+        for (col_idx, col) in row.into_iter().enumerate() {
+            output[col_idx][row_idx] = col;
+        }
+    }
```

Comment:
If you'd just like to get a `Vec<Replicated<F>>' representing the nth bit of all the rows, you can do:

```
let nth_bit:Vec<Replicated<F>> = converted_shares.iter().map(|bitwise_shares| {
    bitwise_shares[n]
}).collect();
```